.github/workflows/github-actions.yml

```diff
@@ -0,0 +1,65 @@
+name: Tests
+on:
+  push:
+    branches: '**'
+  pull_request:
+    branches: main
+
+jobs:
+  unit-testing:
+    defaults:
+      run:
+        shell: bash -el {0}
+
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+    
+    steps:
+      - name: Install xmllint on Linux
+        if: runner.os == 'Linux'
+        run: sudo apt-get install libxml2-utils
+            
+      - name: Get safe PYTHON_VERSION
+        run:  echo "PYTHON_VERSION=$(curl --location https://devguide.python.org/versions | xmllint --html --xpath '//section[@id="supported-versions"]//table/tbody/tr[count(//section[@id="supported-versions"]//table/tbody/tr[td[.="security"]]/preceding-sibling::*)]/td[1]/p/text()' - 2> /dev/null)" >> $GITHUB_ENV
+  
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+
+      - name: Update environment.yml with PYTHON_VERSION
+        run: sed 's/python.*/python==${{ env.PYTHON_VERSION }}/' environment.yml > environment_new.yml
+       
+      - name: Display environment_new.yml
+        run: cat environment_new.yml
+       
+      - name: set up conda and install dependencies
+        uses: conda-incubator/setup-miniconda@v3
+        with:
+          environment-file: environment_new.yml
+        
+      - name : Remove environment_new.yml
+        run: rm -rf environment_new.yml
+       
+      - name: install pytest and coverage
+        run: |
+          conda install conda-forge::pytest
```

Comment:
Why not put these inside of your `environment.yml`?

Reply:
To be consistent with [the environment.yml file in stumpy](https://github.com/TDAmeritrade/stumpy/blob/main/environment.yml), I will add those two libraries to `environment.yml `.

----

I wasn't sure about the **scope** of the existing `environment.yml`, available in the `main` branch. It seems that it contains all the libraries that are required for running the program and developing it. So, on one hand, I thought `pytest` and `coverage` are only for testing, and maybe should be kept outside of `environment.yml`.

Reply:
I think that since it isn't being used to build a package that we should be okay